components/brave_today/browser/direct_feed_controller.cc

```diff
@@ -72,6 +75,117 @@ DirectFeedController::DirectFeedController(
 
 DirectFeedController::~DirectFeedController() = default;
 
+void DirectFeedController::FindFeeds(
+    const GURL& possible_feed_or_site_url,
+    mojom::BraveNewsController::FindFeedsCallback callback) {
+  // Download and check headers. If it's an html document, then parse for rss
+  // items
+  auto request = std::make_unique<network::ResourceRequest>();
+  request->url = possible_feed_or_site_url;
+  request->load_flags = net::LOAD_DO_NOT_SAVE_COOKIES;
+  request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  request->method = net::HttpRequestHeaders::kGetMethod;
+  auto url_loader = network::SimpleURLLoader::Create(
+      std::move(request), GetNetworkTrafficAnnotationTag());
+  url_loader->SetRetryOptions(
+      1, network::SimpleURLLoader::RetryMode::RETRY_ON_5XX |
+             network::SimpleURLLoader::RetryMode::RETRY_ON_NETWORK_CHANGE);
+  url_loader->SetAllowHttpErrorResults(true);
+  auto iter = url_loaders_.insert(url_loaders_.begin(), std::move(url_loader));
+  iter->get()->DownloadToString(
+      url_loader_factory_.get(),
+      // Handle response
+      base::BindOnce(
+          [](DirectFeedController* direct_feed_controller,
+             SimpleURLLoaderList::iterator iter,
+             mojom::BraveNewsController::FindFeedsCallback callback,
+             const GURL& feed_url,
+             const std::unique_ptr<std::string> response_body) {
+            VLOG(1) << "1";
+            VLOG(1) << "Download complete of " << feed_url.spec();
+            // Parse response data
+            auto* loader = iter->get();
+            auto response_code = -1;
+            std::string mime_type;
+            GURL final_url(loader->GetFinalURL());
+            base::flat_map<std::string, std::string> headers;
+            if (loader->ResponseInfo()) {
+              auto headers_list = loader->ResponseInfo()->headers;
+              mime_type = loader->ResponseInfo()->mime_type;
+              if (headers_list) {
+                response_code = headers_list->response_code();
+              }
+            }
+            direct_feed_controller->url_loaders_.erase(iter);
+            std::vector<mojom::FeedSearchResultItemPtr> results;
+            // Check valid response
+            std::string body_content = response_body ? *response_body : "";
+            if (response_code < 200 || response_code >= 300 ||
```

Comment:
I think you could simplify this to only accepting `200` response codes. The other `2xx` are unlikely to come with a valid page / RSS link.

Reply:
done